### PR TITLE
#75 added feature (into endpoint which generates drawing results) to send emails with drawing result to each person…

### PR DESCRIPTION
**Pull request by adrianczekaj**
… who has email

### CODE REVIEW

File: GiftGivingGenerator.API/Controllers/DrawingResultsController.cs [hunk 1]
```diff
@@ -24,15 +33,42 @@ public ActionResult GenerateDrawingResults([FromRoute] Guid eventId)
 		var @event = _eventRepository.Get(eventId);
 		var numberOfTries = @event.DrawResultsAndNumberTries();
 		Log.Information($"For event {eventId} I trie {numberOfTries} times to draw result.");
-
 		_eventRepository.Update(@event);
+		
+		var organizer = _personRepository.Get(@event.OrganizerId);
```
Comment (maciejczekaj) [inline GiftGivingGenerator.API/Controllers/DrawingResultsController.cs]: get all required data from repository using projection

File: GiftGivingGenerator.API/Controllers/DrawingResultsController.cs [hunk 2]
```diff
@@ -24,15 +33,42 @@ public ActionResult GenerateDrawingResults([FromRoute] Guid eventId)
...
+		var personsWithEmail = @event.Persons
+			.Where(x=>x.Email!=null);
+		
+		foreach (var person in personsWithEmail)
+		{
+			var id = _drawingResultRepository.GetByPerson(person.Id).Id;
+			Console.WriteLine(person.Name);
```
Comment (maciejczekaj) [inline GiftGivingGenerator.API/Controllers/DrawingResultsController.cs]: Remove all WriteLines

File: GiftGivingGenerator.API/Controllers/DrawingResultsController.cs [hunk 3]
```diff
@@ -24,15 +33,42 @@ public ActionResult GenerateDrawingResults([FromRoute] Guid eventId)
...
+			Console.WriteLine(organizer.Name);
+			Console.WriteLine(@event.Name);
+
+			var body = $@"Hello {person.Name},
```
Comment (maciejczekaj) [inline GiftGivingGenerator.API/Controllers/DrawingResultsController.cs]: change to `<p>`

File: GiftGivingGenerator.API/Controllers/DrawingResultsController.cs [hunk 4]
```diff
@@ -9,13 +12,19 @@ namespace GiftGivingGenerator.API.Controllers;
 [Route("[controller]")]
 public class DrawingResultsController : ControllerBase
 {
-	private readonly IDrawingResultRepository _repository;
+	private readonly IPersonRepository _personRepository;
+	private readonly IDrawingResultRepository _drawingResultRepository;
 	private readonly IEventRepository _eventRepository;
+	private readonly AppSettings _settings;
+	private readonly IMailService _mail;
```
Comment (maciejczekaj) [inline GiftGivingGenerator.API/Controllers/DrawingResultsController.cs]: mailService

**Review by maciejczekaj — CHANGES_REQUESTED** [review]

File: GiftGivingGenerator.API/Controllers/DrawingResultsController.cs (hunk 1 above)
Comment (adrianczekaj) [reply GiftGivingGenerator.API/Controllers/DrawingResultsController.cs]: Can be it in this way? - see commit

**Review by maciejczekaj — APPROVED** [review]

**Pull request merged**